src/main/scala/za/co/absa/abris/avro/read/confluent/SchemaManager.scala

```diff
@@ -50,12 +63,34 @@ object SchemaManager {
     *
     * This method returns the subject name based on the topic and to which part of the message it corresponds.
     */
-  def getSubjectName(topic: String, isKey: Boolean): String = {
-    if (isKey) {
-      topic + "-key"
-    } else {
-      topic + "-value"
+  def getSubjectName(topic: String, isKey: Boolean, schema: Schema, params: Map[String, String]): String = {
+    val adapter = getSubjectNamingStrategyAdapter(isKey, params)
+
+    if (adapter.validate(schema)) {
+      val subjectName = adapter.subjectName(topic, isKey, schema)
+      logger.info(s"Subject name resolved to: $subjectName")
+      subjectName
     }
+    else {
+      logger.error(s"Invalid configuration for naming strategy. Are you using RecordName or TopicRecordName? " +
+        s"If yes, are you providing SchemaManager.PARAM_SCHEMA_NAME_FOR_RECORD_STRATEGY and " +
+        s"SchemaManager.PARAM_SCHEMA_NAMESPACE_FOR_RECORD_STRATEGY in the configuration map?")
+      null
```

Comment:
I assume this is null so we don't break backward compatibility by making the return type `Option[String]`, that's fine